created first entity - Book

Hryniv opened the pull request:
created:
- Book class
- Book's dao and service layers 
- DataProcessingException for dao layer
- tests
- aplication.properties files for tests and main program

Commit:
created Book class with Lombok and Data JPA annotations

Commit:
created DataprocessingException class for Dao Layer

Commit:
added BookRepository interface with it implementation for Book entity

Commit:
added BookService interface with it implementation for Book entity

Commit:
updated application.properties configuration

- Added database configuration settings

Commit:
updated BookStoreApplication

Commit:
added h2 dependency for test properties

Commit:
added properties for tests

Commit:
added tests

Commit:
code style fixes

Elena-Bruyako commented on src/main/java/com/bookstore/BookStoreApplication.java:
```diff
@@ -1,13 +1,34 @@
+    @Bean
+    public CommandLineRunner commandLineRunner() {
+        return args -> {
+            //Book book = new Book();
```
Uncomment code

Elena-Bruyako commented on src/main/java/com/bookstore/model/Book.java:
```diff
@@ -0,0 +1,27 @@
+import lombok.Data;
+
+@Entity
+@Data
```
Don't use `@Data` for entities
Add `@Table` with name

Elena-Bruyako reviewed (changes requested):
Good job! Let's improve your solution

Commit:
updated Book class:
- deleted @Data annotation
- added @Table annotation and getters, setters, equals, hashCode, toString methods

Commit:
code style fixes in Book class

Commit:
updated BookStoreAplication class:
- uncomented the code in the commandLineRunner method

Commit:
updated BookStoreApplicationTests class:
- due to uncommenting the code in the BookStoreApplication file, the value of the expected indices in the tests was changed
- the Kobzar constant has been changed

Elena-Bruyako commented on src/main/java/com/bookstore/model/Book.java:
```diff
@@ -0,0 +1,123 @@
+    private String description;
+    private String coverImage;
+
+    public Book() {
```
Read which annotations `@Data` includes and use required

Elena-Bruyako reviewed (changes requested):
see comment

Commit:
updated Book class:
- replaced boilerplate code with @Getter, @Setter, @ToString,
@EqualsAndHashCode annotations from lombok
